tests: benchmark.kernel.footprints.pm: all NXP ke series build failure

### Describe the bug

- frdm_ke15z
- frdm_ke17z/ke17z512
- twr_ke18f

eb785ef63b75867f71624c16dafa787757ddc6d0 is the first bad commit
commit eb785ef63b75867f71624c16dafa787757ddc6d0
Author: Felix Wang <fei.wang_3@nxp.com>
Date:   Fri Aug 29 18:25:40 2025 +0800

    drivers: Counter: Fix LPTMR prescaler setting not work issue
    
    In current code, if timer-mode-sel is 0(Timer Counter Mode),
    bypass_prescaler_glitch will be 1, that makes prescaler setting
    be bypassed. But this setting is very useful, especially for
    long timing count.
    
    In this patch, we update prescale-glitch-filter default value to 0,
    to indicate prescaler and glitch filter are disabled, which comply
    the existing devices DTS setting (prescaler = 1).
    And if user sets prescale-glitch-filter properity other than 0,
    we should set bypass_prescaler_glitch to false to make prescaler work,
    and the clock frequence should be calculated with prescaler setting.
    
    Update prescaler field in dt-bindings to tell developer should use
    prescale-glitch-filter instead.
    
    Signed-off-by: Felix Wang <fei.wang_3@nxp.com>


### Regression

- [x] This is a regression.

### Steps to reproduce

```
west build -b frdm_ke15z tests/benchmarks/footprints -- -DCONF_FILE=prj_pm.conf
```

### Relevant log output

```shell
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/timer/mcux_lptmr_timer.c:45:25:
note: in expansion of macro 'TO_LPTMR_PRESCALER' 45 | #define LPTMR_PRESCALER TO_LPTMR_PRESCALER(DT_INST_PROP(0, prescaler)); | ^~~~~~~~~~~~~~~~~~/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/timer/mcux_lptmr_timer.c:
113:24: note: in expansion of macro 'LPTMR_PRESCALER' 113 | config.value = LPTMR_PRESCALER; | ^~~~~~~~~~~~~~~/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/timer/mcux_lptmr_timer.c:
37:44: note: each undeclared identifier is reported only once for each function it appears in 37 | #define TO_LPTMR_PRESCALER(val) _DO_CONCAT(LPTMR_PRESCALER_, val) | ^~~~~~~~~~~~~~~~/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/toolchain/common.h:
168:26: note: in definition of macro '_DO_CONCAT' 168 | #define _DO_CONCAT(x, y) x ## y | ^/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/timer/mcux_lptmr_timer.c:45:25: note: in expansion of macro 'TO_LPTMR_PRESCALER' 45 | #define LPTMR_PRESCALER TO_LPTMR_PRESCALER(DT_INST_PROP(0, prescaler)); | ^~~~~~~~~~~~~~~~~~/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/timer/mcux_lptmr_timer.c:
113:24: note: in expansion of macro 'LPTMR_PRESCALER' 113 | config.value = LPTMR_PRESCALER; | ^~~~~~~~~~~~~~~ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: ubuntu
- toolchain: Zephyr
- version: v4.2.0-6134-g6b8fcaeecf95

### Additional Context

_No response_

--- a/drivers/timer/mcux_lptmr_timer.c
+++ b/drivers/timer/mcux_lptmr_timer.c
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 2021 Vestas Wind Systems A/S
+ * Copyright (c) 2025 NXP
  *
  * SPDX-License-Identifier: Apache-2.0
  */
@@ -17,33 +18,13 @@
 BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
 	     "No LPTMR instance enabled in devicetree");
 
-/* Prescaler mapping */
-#define LPTMR_PRESCALER_2     kLPTMR_Prescale_Glitch_0
-#define LPTMR_PRESCALER_4     kLPTMR_Prescale_Glitch_1
-#define LPTMR_PRESCALER_8     kLPTMR_Prescale_Glitch_2
-#define LPTMR_PRESCALER_16    kLPTMR_Prescale_Glitch_3
-#define LPTMR_PRESCALER_32    kLPTMR_Prescale_Glitch_4
-#define LPTMR_PRESCALER_64    kLPTMR_Prescale_Glitch_5
-#define LPTMR_PRESCALER_128   kLPTMR_Prescale_Glitch_6
-#define LPTMR_PRESCALER_256   kLPTMR_Prescale_Glitch_7
-#define LPTMR_PRESCALER_512   kLPTMR_Prescale_Glitch_8
-#define LPTMR_PRESCALER_1024  kLPTMR_Prescale_Glitch_9
-#define LPTMR_PRESCALER_2048  kLPTMR_Prescale_Glitch_10
-#define LPTMR_PRESCALER_4096  kLPTMR_Prescale_Glitch_11
-#define LPTMR_PRESCALER_8192  kLPTMR_Prescale_Glitch_12
-#define LPTMR_PRESCALER_16384 kLPTMR_Prescale_Glitch_13
-#define LPTMR_PRESCALER_32768 kLPTMR_Prescale_Glitch_14
-#define LPTMR_PRESCALER_65536 kLPTMR_Prescale_Glitch_15
-#define TO_LPTMR_PRESCALER(val) _DO_CONCAT(LPTMR_PRESCALER_, val)
-
 /* Prescaler clock mapping */
 #define TO_LPTMR_CLK_SEL(val) _DO_CONCAT(kLPTMR_PrescalerClock_, val)
 
 /* Devicetree properties */
 #define LPTMR_BASE ((LPTMR_Type *)(DT_INST_REG_ADDR(0)))
-#define LPTMR_CLK_SOURCE TO_LPTMR_CLK_SEL(DT_INST_PROP(0, clk_source));
-#define LPTMR_PRESCALER TO_LPTMR_PRESCALER(DT_INST_PROP(0, prescaler));
-#define LPTMR_BYPASS_PRESCALER DT_INST_PROP(0, prescaler) == 1
+#define LPTMR_CLK_SOURCE TO_LPTMR_CLK_SEL(DT_INST_PROP_OR(0, clk_source, 0))
+#define LPTMR_PRESCALER BIT(DT_INST_PROP_OR(0, prescale_glitch_filter, 0))
 #define LPTMR_IRQN DT_INST_IRQN(0)
 #define LPTMR_IRQ_PRIORITY DT_INST_IRQ(0, priority)
 
@@ -100,18 +81,12 @@ static int sys_clock_driver_init(void)
 {
 	lptmr_config_t config;
 
-
 	LPTMR_GetDefaultConfig(&config);
 	config.timerMode = kLPTMR_TimerModeTimeCounter;
 	config.enableFreeRunning = false;
 	config.prescalerClockSource = LPTMR_CLK_SOURCE;
-
-#if LPTMR_BYPASS_PRESCALER
-	config.bypassPrescaler = true;
-#else /* LPTMR_BYPASS_PRESCALER */
-	config.bypassPrescaler = false;
+	config.bypassPrescaler = !LPTMR_PRESCALER;
 	config.value = LPTMR_PRESCALER;
-#endif /* !LPTMR_BYPASS_PRESCALER */
 
 	LPTMR_Init(LPTMR_BASE, &config);
 

